src/link/index.ts

```diff
@@ -128,6 +130,46 @@ export default async function linkPackages (
       pkgId: pkg.id,
     })
   }
+  if (opts.shamefullyFlatten) {
+    const compatiblePkgsToLinkByName = flatResolvedDeps
+      .filter(pkg => pkg.depth === 0)
```

Comment:
aren't these already symlinked? `depth === 0` are the direct deps of the project